src/main/fc/rc.c

```diff
@@ -405,8 +406,8 @@ uint16_t getCurrentRxRefreshRate(void)
 FAST_CODE_NOINLINE int calcRcSmoothingCutoff(int avgRxFrameTimeUs, bool pt1, uint8_t autoSmoothnessFactor)
 {
     if (avgRxFrameTimeUs > 0) {
-        const float cutoffFactor = (100 - autoSmoothnessFactor) / 100.0f;
-        float cutoff = (1 / (avgRxFrameTimeUs * 1e-6f)) / 2;  // calculate the nyquist frequency
+        const float cutoffFactor = 1.0f / (1.0f + (autoSmoothnessFactor / 10.0f));
+        float cutoff = (1 / (avgRxFrameTimeUs * 1e-6f));  // calculate the Rx frequency
```

Comment:
This logic doesn't limit the filter cutoff frequency to the nyquist of the input signal as the previous correct logic did. Trying to set a lowpass above the nyquist won't work.

Reply:
Nyquist limits only have relevance to the sampling or data rate.  
The lowpass filter is smoothing a set of square inputs which are of much lower frequency than the frequency at which they are sampled and filtered, ie the PID loop rate.
Running a lowpass at the same frequency as the step rates just doesn't round them off very much, that's all.
Here is an image showing a 4k PID loop running a lowpass at 500hz over a 250hz set of steps, ie steps that occur every 4ms.
This is an exact emulation of the current Betaflight code, same maths exactly.
There isn't much smoothing, as expected, but nothing bad happens, either.

![Screen Shot 2021-03-26 at 12 34 47](https://user-images.githubusercontent.com/11737748/112564746-efa35d00-8e2f-11eb-836a-1240207fe90c.jpg)

Reply:
@ctzsnooze A little off topic but why don't we just adjust Q to get rid of the "biquad spike"? With Q = 0.5 damping factor and gain is 1 and there's no overshoot. Like this. I don't see why we would want to use a filter where the step response overshoots for the discrete signals we're dealing with.
![Figure_1](https://user-images.githubusercontent.com/41271048/112658612-dc09ed80-8e53-11eb-8bda-d5d28123078d.png)
![Figure_2](https://user-images.githubusercontent.com/41271048/112658615-dca28400-8e53-11eb-9024-8949ea004871.png)

Reply:
That's a great idea!
I had long been worried about these overshoots on biquad, but had no clue how to eliminate them.  
No-one wants those overshoots.
It seems to change the functional 'strength' of the filter a bit, though.
It's perfect when the cutoff equals the step frequency, which is the new default, but appears 'stronger' than before when the cutoff is lower, with more delay for a given 'smoothness'.

This diagram compares Q of 0.707 on left with 0.5 on right, at the same cutoff frequencies:

![Screen Shot 2021-03-27 at 10 16 23](https://user-images.githubusercontent.com/11737748/112701780-a7467680-8ee5-11eb-895e-5bf779317f00.jpg)

Clearly on the right, with Q = 0.5, we get about the same smoothing, but lag more.

The centre panel is the default behaviour, which would be a fraction more delayed than before, but with the same smoothness, if we used Q = 0.5.

If we adjust the cutoff so the lag is about the same, we get less smoothing....

![Screen Shot 2021-03-27 at 10 16 16](https://user-images.githubusercontent.com/11737748/112701837-d5c45180-8ee5-11eb-9561-42855133d89c.jpg)

So it would seem that if we eliminate the overshoot, we get a bit more delay for the same smoothing.

Hmm.  Have to think about what matters most :-)

Reply:
Here I use high smoothing, imagine it is on a fast RC link where the delay doesn't matter so much, but the pilot wants a smooth output.
With Q = 0.707, the un-commanded overshoot happens almost six RC steps after the sticks are steadied.
This is so late as to be unacceptable:

![Screen Shot 2021-03-27 at 10 27 00](https://user-images.githubusercontent.com/11737748/112702464-aa426680-8ee7-11eb-994d-f0d1f5ee3ec2.jpg)

With Q = 0.5, there is more delay, but never any of that overshoot.
![Screen Shot 2021-03-27 at 10 30 41](https://user-images.githubusercontent.com/11737748/112702485-bb8b7300-8ee7-11eb-8abc-d0812b9945ff.jpg)

If we reduce the smoothing factor a bit, to 50 instead of 70, we can get the same 'peak' value for the input, with a slightly less smooth, but still adequately smooth, trace.

![Screen Shot 2021-03-27 at 10 27 23](https://user-images.githubusercontent.com/11737748/112702480-b5959200-8ee7-11eb-819c-64305c01094a.jpg)

Overall, would people here agree to change the Q factor to 0.5 to prevent the overshoot, even if it makes the smoothing a little bit 'stronger' in terms of delay?

Reply:
Biquad with Q = 0.5 is equivalent to two cascaded first order filters. I think your examples above are pretty extreme with filter cutoffs at half and one fourth of the step frequency.  
The use of filtering in RC smoothing is special. We're deliberately using "too little" filtering to reduce latency so that the filtered signal is really just a series of step responses. 
As we see in the examples above, when the biquad with Q = 0.707 is used there is overshoot that increases with filter cutoff. 
We also see that there is a very sharp transition between steps when a PT1 is used. 
I think we should consider replacing both of these filters with biquad with Q = 0.5 for rc smoothing as this filter doesn't have the overshoot and it has a smoother transition between steps. It would also simplify things for the user.
The cutoff can still be adjusted to get as little delay as one might want. Without overshoot and still with a smoother transition than PT1. I think reducing the smoothing factor like you suggest is a good idea! 👍

Reply:
This auto rc_smoothing cutoff code has been incorporated into #10650, which once merged makes this redundant, so I'll remove it from here.